### PR TITLE
Switch to github actions for CI/CD

### LINKED ISSUE #9: Switch from circleCI to github actions
Package publishing to npm seems to be broken.
Match the CI/CD pipeline we are using on other repos by switching to github actions.

### PR DESCRIPTION
**Pull request by warrenv**
Remove circleCI config and add github actions config.

Closes #9

### CODE REVIEW

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+on: pull_request
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        token: '${{ secrets.PAT_DECRAPIFIER }}'
```
Comment (rudolph9) [inline .github/workflows/pull-request.yaml]: We only need this if we're going to run duti stuff. You should be able to do:
```
    - name: Checkout
      uses: actions/checkout@v2
```

A little extra background on `PAT_DECRAPIFIER` it's only needed since duti pushes up commits which you can't do with the default github token in the secretes.

File: .github/workflows/pull-request.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      uses: actions/cache@v2
+      with:
+        # npm cache files are stored in `~/.npm` on Linux/macOS
+        path: ~/.npm
+        key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.OS }}-node-
+          ${{ runner.OS }}-
+    - run: npm ci
+    - run: npm run lint:ci || true
```
Comment (rudolph9) [inline .github/workflows/pull-request.yaml]: This `|| true` is only so duti can report the results. Without duti these will silently fail. You can get rid of it here.

On that note, there is an edge case in our public repos where if the lint or test scripts fail for a reason that doesn't result in an `results.json` file, the steps will silently fail. I'll create a ticket to clean that up.

**Review by rudolph9 — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-request.yaml (hunk 2 above)
Comment (warrenv) [reply .github/workflows/pull-request.yaml]: done.

File: .github/workflows/pull-request.yaml (hunk 1 above)
Comment (warrenv) [reply .github/workflows/pull-request.yaml]: done.

**Review by rudolph9 — APPROVED** [review]
